### DISCUSSION TITLE
Avoid warning for deprecated rule without replacement

### DISCUSSION THREAD

**Discussion by aaime** [pmd/pmd#3800] [Q&A] [answered]
I'm working on upgrading PMD from 6.35 to 6.41.

Looking at the build scroll, I have found deprecated rules warnings, and replaced them with newer equivalents... until I've stumbled into this:

``
[WARNING] Discontinue using Rule name category/java/performance.xml/UnnecessaryWrapperObjectCreation as it is scheduled for removal from PMD. PMD 7.0.0 will remove support for this Rule.
``

I go look for the replacement and find this in the documentation:

> Deprecated since PMD 6.37.0. The planned replacement is not expected before PMD 7.0.0.

I don't want to remove the check, and yet, I cannot have a warning message pop up about it for each one of the 100+ Maven modules that the build works through. Is there any way to quiet that message?

**Comment by oowekyala** [comment, marked as answer]
For this specific rule, my suggestion would be to remove it from your ruleset. It only flags expressions like `Integer.valueOf(1).intValue()`, ie where the `intValue` is called *immediately* on a `valueOf` in the same expression. This is incredibly rare, contrived even, so I don't think you'd be missing any functionality by disabling the rule. The replacement rule in PMD 7 is much more powerful, it might as well be an unrelated rule.
